receiver/googlecloudspannerreceiver/internal/metadataparser/metadataconfig.go

```diff
@@ -0,0 +1,26 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadataparser
+
+type MetadataType int32
```

Comment:
This type can be moved to `MetricsMetadata` and it can leverage be exposing method like

```
func (metadata *MetricsMetadata) MetadataType() MetadataType {
	if metadata.TimestampColumnName == "" {
		return MetadataTypeCurrentStats
	}
	return MetadataTypeIntervalStats
}

``` 

By having this method there no need anymore to maintain maps per each type eg for `DatabaseReader`

```
func initializeReaders(logger *zap.Logger, parsedMetadata []*metadata.MetricsMetadata,
	database *datasource.Database, readerConfig ReaderConfig) []Reader {
	var readers []Reader
	for _, m := range parsedMetadata {
		switch m.MetadataType() {
		case metadata.MetadataTypeCurrentStats:
			readers = append(readers, newCurrentStatsReader(logger, database, m, readerConfig))
		case metadata.MetadataTypeIntervalStats:
			readers = append(readers, newIntervalStatsReader(logger, database, m, readerConfig))
		}
	}
	return readers
}

```

Reply:
Done